08_Scaling_to_multiple_GPUs/README.md

```diff
@@ -0,0 +1,83 @@
+
+   However, we are using the HuggingFace Trainer which will automatically take care of setting up data-parallel training for the model if it detects the environment variables set by torchrun, so we do not need to handle that manually.
+
+   **Task**: You will need to make the following changes to the script:
+   - select the correct PyTorch device
+   - adjust the per-device batch size handled per process
+   - limit printing of outputs to a single process
+  
+   Places where you need to edit the file have been marked with `<!!! ACTION REQUIRED ... !!!>`.
+  
+2. Adjust the slurm batch file.
+
+   **Task**: Edit the slurm batch file [08_Scaling_to_multiple_GPUs/run.sh](08_Scaling_to_multiple_GPUs/run.sh) for single-node multi-gpu training using torchrun.
+
+   You should specify at least the following:
+    - the correct slurm partition
+    - number of GPUs requested (8)
+    - number of CPUs requested
+    - RAM requested (you can use `0` to reserve all memory on the node)
+    - requested runtime
+    - the course reservation
+
+   It can also be helpful to specify a name for the slurm logfile that contains the command line outputs of the script.
+
+   You will also need to add the relevant parts for setting up the PyTorch software environment (these are the same as for Exercise `03_Your_first_AI_training_job_on_LUMI`).
+
+   To invoke torchrun from the batch file, follow the [Single-node multi-worker usage example on the torchrun website](https://pytorch.org/docs/stable/elastic/run.html#single-node-multi-worker).
+
+   **Task**: Run your job using `sbatch run.sh`.
+
+   **Task**: Compare how the run time differs between running on a full node and the previous run on a single GCD.
+
+3. Set up CPU bindings.
+
+   In order to achieve optimal CPU-GPU data transfer performance we need to ensure that each script remains on the CPU cores closest to the respective GPU.
+   As we are using torchrun to manage the worker processes, we cannot handle these CPU bindings via slurm but must set them up in our Python training script.
+
+   **Task**: Edit [08_Scaling_to_multiple_GPUs/GPT-neo-IMDB-finetuning.py](08_Scaling_to_multiple_GPUs/GPT-neo-IMDB-finetuning.py) to set up the correct CPU-GPU bindings based on the processes rank.
+
+4. (Optional/Bonus): Running without PyTorch.
+
+   We can also start worker processes directly without using torchrun to have direct control over all processes.
+   
+   **Task**: Change the slurm batch script to
+   - instruct slurm to start the appropriate number of processes,
+   - set the environment variables mentioned above manually,
+   - replace the `torchrun` invocation with direct `python` commands to run the training script.
+   
+   > **Note**
+   >
+   > You can get the hostname of the node running the rank 0 process using the command
+   > ```
+   > scontrol show hostname ${SLURM_NODELIST} | head -n 1
```

Comment:
I think a simple `hostname`, maybe `hostname -i`, is sufficient here (as you suggested in https://docs.csc.fi/support/tutorials/ml-multi/). It worked in my examples at least...